File: backend/src/main/java/ibee/webapp/todo_app/core/dto/person/contact/mail/PersonEmailAddressDto.java
package ibee.webapp.todo_app.core.dto.person.contact.mail;

import ibee.webapp.todo_app.core.dto.EmailAddressDto;
import ibee.webapp.todo_app.features.person.related.referenceIds.contact.PersonEmailAddressDtoId;
import jakarta.validation.constraints.NotNull;

public record PersonEmailAddressDto(

    @NotNull
    PersonEmailAddressDtoId id,

    @NotNull
    EmailAddressDto emailAddress,

    @NotNull
    Boolean mainEmail
) {

}

File: frontend (2)/src/app/types/dto/person/related/contact/mail/person-email-address.dto.ts
import { EmailAddressDto } from '../../../../email-address.dto';
import { PersonEmailAddressDtoId } from '../../reference/contact/person-email-address-dto-id';
import { PersonPhoneNumberDtoId } from '../../reference/contact/person-phone-number-dto-id';


export interface PersonEmailAddressDto {

    id: PersonEmailAddressDtoId;

    emailAddress: EmailAddressDto;

    mainEmail: boolean;

}